frontend/src/app/share/[threadId]/layout.tsx

```diff
@@ -1,19 +1,46 @@
 import { Metadata } from 'next';
```

Comment:
Explanation: 

- here `thread` and the corresponding `project` are fetched to pass the project title and description in the page's meta title and description and open graph tags. This is server-side and right place to set the meta tags
- we use the route `/api/share-page/og-image/route.tsx` to generate OG image based on the project's title